### PR TITLE
Module3

### PR DESCRIPTION
**Pull request by LaiqianDS**
Milestone 1 is done
I haven't started milestone 2 yet, but I wanted to check the first part

### CODE REVIEW

File: src/module_3/models/20241104193743_logistic_regression.pkl
Comment (fsanzestevez) [inline src/module_3/models/20241104193743_logistic_regression.pkl:1]: I don't know if this is the best place to store binaries of a model.. 
We should avoid this kind of uploads to a repository (same as with data, API Keys, etc).

File: src/module_3/models/20241104194930_logistic_regression.pkl
Comment (fsanzestevez) [inline src/module_3/models/20241104194930_logistic_regression.pkl:1]: Same as above

File: src/module_3/predict.py
Comment (fsanzestevez) [inline src/module_3/predict.py:1]: Awesome idea to make it modular!!

File: src/module_3/predict.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from joblib import load
+from train import OUTPUT_PATH, evaluate_model, clean_dataset
+import pandas as pd
+
+def main():
+    model_name = '/20241104193743_logistic_regression.pkl'
```
Comment (fsanzestevez) [inline src/module_3/predict.py]: Maybe we could make an input instead of hardcoding the model binaries. This doesn't take much advantage of reproducibility we get from a modular implementation

File: src/module_3/train.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import pandas as pd
+import numpy as np
+import datetime
+import joblib
+
+from functions import three_way_split_time, get_best_treshold, pick_features_for_splits, logit_significance
+
+OUTPUT_PATH = './src/module_3/models'
+
+def clean_dataset(df: pd.DataFrame) -> pd.DataFrame:
```
Comment (fsanzestevez) [inline src/module_3/train.py:21]: Awesome, using typing!

File: src/module_3/train.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+from sklearn.base import BaseEstimator
+from sklearn.metrics import precision_recall_curve, auc, roc_auc_score
+import statsmodels.api as sm
+from statsmodels.discrete.discrete_model import LogitResults
+import pandas as pd
+import numpy as np
+import datetime
+import joblib
+
+from functions import three_way_split_time, get_best_treshold, pick_features_for_splits, logit_significance
```
Comment (fsanzestevez) [inline src/module_3/train.py]: What is `functions`? Did we miss a file in the PR? 
I don't see much here: https://pypi.org/project/functions/

**Review by fsanzestevez — APPROVED** [review]
I'm missing conclusions. 
I think there is one file missing `functions` . 
Good modularity, good typing, clean code, nice comments. But I don't see results 🙃

File: src/module_3/train.py (hunk 3 above)
Comment (LaiqianDS) [reply src/module_3/train.py]: imagen

It's a file I used to write down a few functions. I just send the path to it in the PR
